File: 1364-tuple-with-same-product/1364-tuple-with-same-product.cpp
class Solution {
public:
    int tupleSameProduct(vector<int>& nums) {
        int ans=0;
        map<int,int>mpp;
        for(int i=0;i<nums.size();i++){
            for(int j=i+1;j<nums.size();j++){
                mpp[nums[i]*nums[j]]++;
            }
        }
        for(auto it :mpp){
            if(it.second>=2){
                ans+=(it.second*(it.second-1)/2)*8;
            }
        }
        return ans;
    }
};

File: 1364-tuple-with-same-product/1364-tuple-with-same-product.py
class Solution:
    def tupleSameProduct(self, nums: List[int]) -> int:
        ans=0
        mpp=defaultdict(int)
        for i in range(len(nums)):
            for j in range(i+1,len(nums)):
                mpp[nums[i]*nums[j]]+=1
        for count in mpp.values():
            if count>=2:
                ways = int(count*(count-1)/2)
                ans+=ways*8
                
        return ans

        

